assets/src/editor/events/initialization/with-event-date-entities.js

```diff
@@ -5,24 +5,24 @@ import { withSelect } from '@wordpress/data';
 import { isModelEntityOfModel } from '@eventespresso/validators';
 import { createHigherOrderComponent } from '@wordpress/compose';
 
-const EMPTY_OBJECT = {};
+const DEFAULT_OBJECT = {
+	dateEntities: [],
+	dateEntitiesLoaded: false,
+};
 
 /**
  * withEventDateEntities
  * returns an object containing the following:
  *    dateEntities - an array of datetime entities for the supplied event
  *    dateEntitiesLoaded - boolean true if loading is complete
  *
- * !!! IMPORTANT !!!
- * ONLY USE THIS AFTER THE CALL TO hasFinishedResolution() in the
- * withEvent() DATA HOC has completed and resolved true
- *
  * @function
  */
 const withEventDateEntities = createHigherOrderComponent(
 	withSelect(
-		( select, { eventEntity, eventEntityLoaded } ) => {
+		( select, { eventEntity, eventEntityLoaded, editorInitialized } ) => {
 			if (
+				! editorInitialized &&
```

Comment:
This ensures that we're only ever invoking this particular logic until the editor is initialized (same for the `withEventEntity`, and `withTicketEntitiesForAllDateEntities` hocs in the initialization layer.  If this was not added, then they selectors will continue to query the store state everytime the state changes.  We don't need that in this case.